src/common/row_operations/row_external.cpp

```diff
@@ -0,0 +1,127 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/types/row_operations/row_external.cpp
+//
+//
+//===----------------------------------------------------------------------===//
+#include "duckdb/common/row_operations/row_operations.hpp"
+#include "duckdb/common/types/row_layout.hpp"
+
+namespace duckdb {
+
+void RowOperations::SwizzleColumns(const RowLayout &layout, const data_ptr_t &base_row_ptr, const idx_t &count) {
```

Comment:
Clang-tidy gave me the impression that const references are the cheapest way to pass parameters to a function. This avoids copying the passed parameters into a new variable. I suppose that this is most noticeable in the RowLayout since copying this class is more costly than copying a single variable. I think that the last two arguments do not need to be references, but I don't think it does any harm.

Reply:
const references are most certainly the cheapest way to copy more complex types, but it is essentially equivalent to passing a pointer, which is pushing 8 bytes onto the stack. Since `data_ptr_t` and `idx_t` are 8 bytes by themselves, it is cheaper to pass them by value as you save an (unnecessary) indirection.

See also the [C++ core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#fcall-parameter-passing) on this topic.

The general rule of thumb is "pass-by-value for simple types (numerics, pointers), pass by const reference for more complex types".